pkg/reconciler/dependencybuild/dependencybuild.go

```diff
@@ -246,27 +248,34 @@ func (r *ReconcileDependencyBuild) handleStateAnalyzeBuild(ctx context.Context,
 			Invocations      [][]string
 			EnforceVersion   string
 			IgnoredArtifacts []string
+			ToolVersion      string
+			JavaVersion      string
 		}{}
 
 		if err := json.Unmarshal([]byte(buildInfo), &unmarshalled); err != nil {
 			r.eventRecorder.Eventf(&db, v1.EventTypeWarning, "InvalidJson", "Failed to unmarshal build info for AB %s/%s JSON: %s", db.Namespace, db.Name, buildInfo)
 			return reconcile.Result{}, err
 		}
-		//for now we are ignoring the tool versions
-		//and just using the supplied invocations
+		// for now we are ignoring the tool versions
+		// and just using the supplied invocations
 		buildRecipes := []*v1alpha1.BuildRecipe{}
-		_, maven := unmarshalled.Tools["maven"]
+		_, maven := unmarshalled.Tools["mvn"]
 		_, gradle := unmarshalled.Tools["gradle"]
-		for _, image := range []string{"quay.io/sdouglas/hacbs-jdk11-builder:latest", "quay.io/sdouglas/hacbs-jdk8-builder:latest", "quay.io/sdouglas/hacbs-jdk17-builder:latest"} {
-			for _, command := range unmarshalled.Invocations {
-				if maven {
-					buildRecipes = append(buildRecipes, &v1alpha1.BuildRecipe{Maven: true, Image: image, CommandLine: command, EnforceVersion: unmarshalled.EnforceVersion, IgnoredArtifacts: unmarshalled.IgnoredArtifacts})
+
+		if maven {
+			for _, image := range []string{"quay.io/sdouglas/hacbs-jdk11-builder:latest", "quay.io/sdouglas/hacbs-jdk8-builder:latest", "quay.io/sdouglas/hacbs-jdk17-builder:latest"} {
+				for _, command := range unmarshalled.Invocations {
+					buildRecipes = append(buildRecipes, &v1alpha1.BuildRecipe{Image: image, CommandLine: command, EnforceVersion: unmarshalled.EnforceVersion, IgnoredArtifacts: unmarshalled.IgnoredArtifacts, ToolVersion: unmarshalled.ToolVersion, JavaVersion: unmarshalled.JavaVersion, Maven: true})
 				}
-				if gradle {
-					buildRecipes = append(buildRecipes, &v1alpha1.BuildRecipe{Gradle: true, Image: image, CommandLine: command, EnforceVersion: unmarshalled.EnforceVersion, IgnoredArtifacts: unmarshalled.IgnoredArtifacts})
+			}
+		} else if gradle {
```

Comment:
yeah we don't like this `else if` stuff in golang (I had to learn this same lesson in 2015 when I left the java world and started working on openshift).

switch this and the `if maven` above to 

```
switch {
case maven:
<add the code for maven>
case gradle:
<add the code for gradle>
default:
return an error cause something is wrong with our base setup
}
```

Reply:
Done in f3b832f.

Reply:
looks good @dwalluck thanks ... marking resolved